controllers/chat/initChatFromPost.js

```diff
@@ -54,39 +54,57 @@ const initChatFromPost = async (req, res) => {
+            postId: postId,
+            userId: targetUserModel.user_id
+          },
+          null,
+          true
+        );
+
+        anon_user_info = {
+          anon_user_info_color_code: postAnonUserInfo?.anon_user_info_color_code,
+          anon_user_info_color_name: postAnonUserInfo?.anon_user_info_color_name,
+          anon_user_info_emoji_code: postAnonUserInfo?.anon_user_info_emoji_code,
+          anon_user_info_emoji_name: postAnonUserInfo?.anon_user_info_emoji_name
+        };
+      } else if (source === 'comment') {
+        const reaction = await Getstream.feed.getReactionById(commentId);
+        anon_user_info = {
+          anon_user_info_color_code: reaction?.data.anon_user_info_color_code,
+          anon_user_info_color_name: reaction?.data.anon_user_info_color_name,
+          anon_user_info_emoji_code: reaction?.data.anon_user_info_emoji_code,
+          anon_user_info_emoji_name: reaction?.data.anon_user_info_emoji_name
+        };
+      }
 
       newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_color_code =
-        postAnonUserInfo?.anon_user_info_color_code;
+        anon_user_info?.anon_user_info_color_code;
       newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_color_name =
-        postAnonUserInfo?.anon_user_info_color_name;
+        anon_user_info?.anon_user_info_color_name;
       newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_emoji_code =
-        postAnonUserInfo?.anon_user_info_emoji_code;
+        anon_user_info?.anon_user_info_emoji_code;
       newStateMemberWithAnonInfo[targetUserModel.user_id].anon_user_info_emoji_name =
-        postAnonUserInfo?.anon_user_info_emoji_name;
+        anon_user_info?.anon_user_info_emoji_name;
 
       newStateMemberWithAnonInfo[targetUserModel.user_id].user.name =
-        'Anonymous ' + postAnonUserInfo?.anon_user_info_emoji_name;
+        'Anonymous ' + anon_user_info?.anon_user_info_emoji_name;
       newStateMemberWithAnonInfo[targetUserModel.user_id].user.username =
-        'Anonymous ' + postAnonUserInfo?.anon_user_info_emoji_name;
+        'Anonymous ' + anon_user_info?.anon_user_info_emoji_name;
 
       await ChatAnonUserInfo.create({
         channel_id: newChannel.id,
         my_anon_user_id: userModel.user_id,
         target_user_id: targetUserModel.user_id,
-        anon_user_info_color_code: postAnonUserInfo?.anon_user_info_color_code,
-        anon_user_info_color_name: postAnonUserInfo?.anon_user_info_color_name,
-        anon_user_info_emoji_code: postAnonUserInfo?.anon_user_info_emoji_code,
-        anon_user_info_emoji_name: postAnonUserInfo?.anon_user_info_emoji_name
+        anon_user_info_color_code: anon_user_info?.anon_user_info_color_code,
+        anon_user_info_color_name: anon_user_info?.anon_user_info_color_name,
+        anon_user_info_emoji_code: anon_user_info?.anon_user_info_emoji_code,
+        anon_user_info_emoji_name: anon_user_info?.anon_user_info_emoji_name
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code for creating a new entry in the `ChatAnonUserInfo` table is repeated four times. This can be simplified by using a loop or a mapping function. Here's how you could do it:

```diff
- await ChatAnonUserInfo.create({
-   channel_id: newChannel.id,
-   my_anon_user_id: userModel.user_id,
-   target_user_id: targetUserModel.user_id,
-   anon_user_info_color_code: anon_user_info?.anon_user_info_color_code,
-   anon_user_info_color_name: anon_user_info?.anon_user_info_color_name,
-   anon_user_info_emoji_code: anon_user_info?.anon_user_info_emoji_code,
-   anon_user_info_emoji_name: anon_user_info?.anon_user_info_emoji_name
- });

+ const chatAnonUserInfoData = {
+   channel_id: newChannel.id,
+   my_anon_user_id: userModel.user_id,
+   target_user_id: targetUserModel.user_id,
+ };
+ ['color_code', 'color_name', 'emoji_code', 'emoji_name'].forEach((key) => {
+   chatAnonUserInfoData[`anon_user_info_${key}`] = anon_user_info[`anon_user_info_${key}`];
+ });
+ await ChatAnonUserInfo.create(chatAnonUserInfoData);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->